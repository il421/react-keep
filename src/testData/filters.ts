import { Filters } from "../store/store.types";

export const filter = {
  tag: "YpjypoYGueKBUukRPQ5F",
  search: "Important",
};

export const filters: Filters = {
  tagFilters: ["1"],
  search: "Important",
};
