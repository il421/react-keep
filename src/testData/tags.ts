import { Tag } from "../store/store.types";

export const tags: Tag[] = [
  {
    id: "1",
    value: "Tag1",
  },
  {
    id: "2",
    value: "Tag2",
  },
];
