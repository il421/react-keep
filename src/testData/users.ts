export const user = {
  email: "app.ilya.testing@gmail.com",
  password: "{SECURE_TEST_USER_PASS}2",
  uid: "qIxSzOgxdGcEFgG9JZg9LyWVLH2345678901",
  firstName: "Ilya",
  lastName: "Suglobov",
  url: null,
};
