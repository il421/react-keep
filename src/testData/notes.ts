import { Note } from "../store/store.types";
import { NoteType } from "../components/notes";
import { tags } from "./tags";

export const notes: Note[] = [
  {
    type: NoteType.text,
    id: "1",
    title: "1",
    content: "text",
    tags: [tags[0].id],
    color: "#1",
    important: false,
    archive: false,
    createdAt: 100000,
    updatedAt: 100005,
  },
  {
    type: NoteType.list,
    id: "2",
    title: "2",
    content: [
      {
        id: "2-1",
        content: "dog",
        checked: false,
      },
      {
        id: "2-2",
        content: "cat",
        checked: true,
      },
    ],
    tags: [tags[0].id, tags[1].id],
    color: "#2",
    important: true,
    archive: false,
    createdAt: 100000,
    updatedAt: 100005,
  },
  {
    type: NoteType.image,
    id: "3",
    title: "3",
    content: { text: "text", imageUrl: "imageUrl" },
    tags: [tags[1].id],
    color: "#3",
    important: false,
    archive: true,
    createdAt: 100000,
    updatedAt: 100005,
  },
];

export const newNote: Omit<Note, "id" | "createdAt" | "updatedAt"> = {
  color: "#cbf0f8",
  important: false,
  archive: false,
  tags: [tags[0].id],
  type: NoteType.text,
  content: "Dont forget to make a call",
  title: "Call",
};

export const updatedNote: Omit<Note, "id"> = {
  color: "#cbf0f8",
  important: false,
  archive: false,
  tags: [tags[0].id],
  type: NoteType.text,
  content: "Dont forget to make a call",
  title: "Call",
  createdAt: 54353534,
  updatedAt: 43432432,
};
