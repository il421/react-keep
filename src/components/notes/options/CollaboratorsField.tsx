import React from "react";
import { FlexBox, Option } from "../../ui-components";
import { FlexWrap, JustifyContent, nameOf } from "../../../common";
import { Collaborator, ImageItem, ListItem } from "../../../store/store.types";
import "../../../styles/components/notes/_collaborators-picker.scss";
import { NoteFormValues } from "../notes.types";
import { CheckboxGroupField } from "../../form/CheckboxGroupField";

interface CollaboratorsFieldProps {
  collaborators: Collaborator[];
}
export class CollaboratorsField extends React.PureComponent<
  CollaboratorsFieldProps
> {
  private nameOf = nameOf<NoteFormValues<string | ListItem[] | ImageItem>>();
  private getOptions = (): Option[] => {
    return this.props.collaborators.map<Option>((coll: Collaborator) => ({
      id: coll.uid,
      value: coll.displayName ?? coll.email! ?? coll.uid,
    }));
  };

  render() {
    return (
      <FlexBox
        justifyContent={JustifyContent.spaceBetween}
        vertical
        className="collaborators-picker"
        flexWrap={FlexWrap.wrap}
      >
        <CheckboxGroupField
          name={this.nameOf("collaborators")}
          options={this.getOptions()}
          classNames={{
            option: "collaborators-picker__wrapper",
            checkbox: "collaborators-picker__checkbox",
            label: "collaborators-picker__value",
          }}
        />
      </FlexBox>
    );
  }
}

export default CollaboratorsField;
