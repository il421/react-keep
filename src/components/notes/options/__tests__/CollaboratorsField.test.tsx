import React from "react";
import { mount, ReactWrapper } from "enzyme";
import CollaboratorsField from "../CollaboratorsField";
import { Form } from "react-final-form";
import { collaborators } from "../../../../testData/users";

test("should render CollaboratorsField correctly", () => {
  const wrapper: ReactWrapper = mount(
    <Form initialValues={{}} onSubmit={() => {}}>
      {() => <CollaboratorsField collaborators={collaborators} />}
    </Form>
  );
  expect(wrapper.debug()).toMatchSnapshot();
});
