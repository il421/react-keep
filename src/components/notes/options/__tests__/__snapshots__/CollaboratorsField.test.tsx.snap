// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CollaboratorsField correctly 1`] = `
"<ReactFinalForm initialValues={{...}} onSubmit={[Function: onSubmit]}>
  <CollaboratorsField collaborators={{...}}>
    <FlexBox justifyContent=\\"space-between\\" vertical={true} className=\\"collaborators-picker\\" flexWrap=\\"wrap\\">
      <div style={{...}} className=\\"collaborators-picker\\">
        <CheckboxGroupField name=\\"collaborators\\" options={{...}} classNames={{...}}>
          <Field name=\\"collaborators\\" options={{...}} classNames={{...}} component={[Function: FieldAdapter]}>
            <FieldAdapter input={{...}} meta={{...}} options={{...}} classNames={{...}} render={[undefined]}>
              <CheckboxGroup options={{...}} classNames={{...}} selectedKeys=\\"\\" onChange={[Function: onChange]}>
                <FlexBox justifyContent=\\"flex-start\\" className=\\"collaborators-picker__wrapper\\">
                  <div style={{...}} className=\\"collaborators-picker__wrapper\\">
                    <div className=\\"collaborators-picker__checkbox\\">
                      <input id=\\"BYKMnT7RSfeyrLiBAZwvGLajEPF2\\" value=\\"BYKMnT7RSfeyrLiBAZwvGLajEPF2\\" type=\\"checkbox\\" defaultChecked={false} />
                      <label htmlFor=\\"BYKMnT7RSfeyrLiBAZwvGLajEPF2\\" onClick={[Function: onClick]} />
                    </div>
                    <span className=\\"collaborators-picker__value\\">
                      Ivan
                    </span>
                  </div>
                </FlexBox>
                <FlexBox justifyContent=\\"flex-start\\" className=\\"collaborators-picker__wrapper\\">
                  <div style={{...}} className=\\"collaborators-picker__wrapper\\">
                    <div className=\\"collaborators-picker__checkbox\\">
                      <input id=\\"Q81hnXu9qJM5Y2iTRkun6gVRj0i1\\" value=\\"Q81hnXu9qJM5Y2iTRkun6gVRj0i1\\" type=\\"checkbox\\" defaultChecked={false} />
                      <label htmlFor=\\"Q81hnXu9qJM5Y2iTRkun6gVRj0i1\\" onClick={[Function: onClick]} />
                    </div>
                    <span className=\\"collaborators-picker__value\\">
                      Tany
                    </span>
                  </div>
                </FlexBox>
              </CheckboxGroup>
            </FieldAdapter>
          </Field>
        </CheckboxGroupField>
      </div>
    </FlexBox>
  </CollaboratorsField>
</ReactFinalForm>"
`;
